app/gui2/src/components/GraphEditor/GraphNode.vue

```diff
@@ -244,40 +244,43 @@ const transform = computed(() => {
 })
 
 const startEpochMs = ref(0)
-let startEvent: PointerEvent | null = null
-let startPos = Vec2.Zero
-
-// TODO[ao]: Now, the dragPointer.events are preventing `click` events on widgets if they don't
-// stop pointerup and pointerdown. Now we ensure that any widget handling click does that, but
-// instead `usePointer` should be smarter.
-const dragPointer = usePointer((pos, event, type) => {
-  if (type !== 'start') {
-    const fullOffset = pos.absolute.sub(startPos)
-    emit('dragging', fullOffset)
-  }
-  switch (type) {
-    case 'start': {
-      startEpochMs.value = Number(new Date())
-      startEvent = event
-      startPos = pos.absolute
-      event.stopImmediatePropagation()
-      break
-    }
-    case 'stop': {
+let draggedElement: Element | undefined
+let significantMove = false
+
+const dragPointer = usePointer(
+  (pos, event, type) => {
+    if (type !== 'start') {
       if (
-        Number(new Date()) - startEpochMs.value <= MAXIMUM_CLICK_LENGTH_MS &&
-        startEvent != null &&
-        pos.absolute.distanceSquared(startPos) <= MAXIMUM_CLICK_DISTANCE_SQ
+        !significantMove &&
+        draggedElement &&
+        (Number(new Date()) - startEpochMs.value >= MAXIMUM_CLICK_LENGTH_MS ||
+          pos.relative.lengthSquared() >= MAXIMUM_CLICK_DISTANCE_SQ)
       ) {
-        nodeSelection?.handleSelectionOf(event, new Set([nodeId.value]))
-        handleNodeClick(event)
+        // If this is clearly a drag (not a click), the node itself capture pointer events to
+        // prevent `click` on widgets.
+        draggedElement.setPointerCapture?.(event.pointerId)
```

Comment:
Is it necessary to store `draggedElement`? Looks like using `event.currentTarget` directly here would have the same effect. Or is it different because of existing pointer capture?